libfuzzer/libfuzz_harness.cc

```diff
@@ -42,26 +41,107 @@ int null_printf(FILE* stream, const char* format, ...)
     return 0;
 }
 
-/**
- * @brief Accept an input buffer and size.
- *
- * @param[in] data Pointer to the input buffer.
- * @param[in] size Size of the input buffer.
- * @return -1 if the input is invalid
- * @return 0 if the input is valid and processed.
- */
-int LLVMFuzzerTestOneInput(const uint8_t* data, std::size_t size)
+typedef std::unique_ptr<ubpf_vm, decltype(&ubpf_destroy)> ubpf_vm_ptr;
+
+ubpf_vm_ptr create_ubpf_vm(const std::vector<uint8_t>& program_code)
 {
-    // Assume the fuzzer input is as follows:
-    // 32-bit program length
-    // program byte
-    // test data
+    // Automatically free the VM when it goes out of scope.
+    std::unique_ptr<ubpf_vm, decltype(&ubpf_destroy)> vm(ubpf_create(), ubpf_destroy);
 
-    // Copy memory into a writable buffer.
-    std::vector<uint8_t> memory;
+    if (vm == nullptr) {
+        // Failed to create the VM.
+        // This is not interesting, as the fuzzer input is invalid.
+        // Do not add it to the corpus.
+        return vm;
+    }
+
+    ubpf_toggle_undefined_behavior_check(vm.get(), true);
+
+    char* error_message = nullptr;
+
+    ubpf_set_error_print(vm.get(), null_printf);
+
+    if (ubpf_load(vm.get(), program_code.data(), program_code.size(), &error_message) != 0) {
+        // The program failed to load, due to a validation error.
+        // This is not interesting, as the fuzzer input is invalid.
+        // Do not add it to the corpus.
+        free(error_message);
+        vm.reset();
+        return vm;
```

Comment:
I know that the `reset` will make the pointer `nullptr`, but do you think that it reads better to return `nullptr` here (and below) to make it more clear what is going on?

Reply:
It would need to be: "return {nullptr, ubpf_destroy};"

If that's clear I am happy to make that change...